### PR TITLE
Fix saving in SavedModel format

### PR DESCRIPTION
**Pull request by lgeiger**
This PR fixes saving of larq models in the TensorFlow `SavedModel` format.

### CODE REVIEW

File: larq/quantized_variable_test.py [hunk 1]
```diff
@@ -299,3 +300,15 @@ def f():
             assert evaluate(x) == -2.0
     else:
         assert evaluate(x) == 0.0
+
+
+@pytest.mark.skipif(
+    version.parse(tf.__version__) < version.parse("2"), reason="Requires TensorFlow 2",
+)
```
Comment (lgeiger) [inline larq/quantized_variable_test.py:307]: I guess we could make this work in TF 1.x as well, but ignore the failure for now.

File: larq/quantized_variable_test.py [hunk 2]
```diff
@@ -299,3 +300,16 @@ def f():
...
+@pytest.mark.skipif(version.parse(tf.__version__) == version.parse("2.3.0rc0"))
```
Comment (lgeiger) [inline larq/quantized_variable_test.py:308]: We should fix this in a future PR or open an issue on TensorFlow

File: larq/quantized_variable_test.py [hunk 3]
```diff
@@ -299,3 +300,19 @@ def f():
...
     else:
         assert evaluate(x) == 0.0
+
+
+@pytest.mark.skipif(
+    version.parse(tf.__version__) < version.parse("2"), reason="Requires TensorFlow 2",
+)
+@pytest.mark.skipif(
+    version.parse(tf.__version__) == version.parse("2.3.0rc0"),
+    reason="Currently broken on 2.3.0rc0.",
```
Comment (leonoverweel) [inline larq/quantized_variable_test.py:310]: Should we make an issue tracking the TF 2.3 release to keep track of potential blockers/things to fix like this?

Comment (lgeiger) [reply larq/quantized_variable_test.py:310]: :+1: I will look into this soon and will either make an issue or provide a fix.

Comment (lgeiger) [reply larq/quantized_variable_test.py]: 2c30f2170cc84e8cd839d59018dbf865a0745f62 should fix the incompatibility with TF 2.3

**Comment by lgeiger** [comment]
The codecov failures are expected since some codepaths only get called on specific versions of TensorFlow.

**Pull request merged**